src/components/radio-group/bl-radio-group.ts

```diff
@@ -100,12 +100,25 @@ export default class BlRadioGroup extends FormControlMixin(LitElement) {
     event.preventDefault();
   }
 
+  getFocusedOptionIndex() {
+    return this.focusedOptionIndex;
+  }
+
   connectedCallback(): void {
     super.connectedCallback();
 
     this.tabIndex = 0;
     this.addEventListener('focus', this.handleFocus);
     this.addEventListener('keydown', this.handleKeyDown);
+    this.addEventListener('mouseover', this.handleMouseOver);
```

Comment:
Mouseover event will trigger with every pixel move over a bl-radio-group element, which seems very inefficient. I think, we should question why do we need that focused option index at all instead.

Reply:
The focusedOptionIndex is for keyboard navigation, but if the user clicks the option with the mouse **handleFocus** is calling so it focuses on the first option. Do we need **handleFocus** in here?

Reply:
I used a flag named **isMouseEventFocus** to track whether the focus is due to a mouse event or not. So handleFocus is executed only with keyboard navigation.